backend/src/sms/services/sms-callback.service.ts

```diff
@@ -94,7 +114,65 @@ const parseEvent = async (req: Request): Promise<void> => {
     )
   }
 }
+
+// for future context: we only care about when the twilio callback is a finalized status
+// we don't want to update the status pre-emptively until we receive the finalized status
+// initial state --> UNSENT
+const parseTransactionalEvent = async (req: Request): Promise<void> => {
+  const {
+    MessageStatus: twilioMessageStatus,
+    ErrorCode: twilioErrorCode,
+    MessageSid: messageId,
+  } = req.body
+  if (FINALIZED_STATUS.indexOf(twilioMessageStatus as string) === -1) {
+    return
+  }
+  logger.info({
+    message: 'Updating message in sms_messages_transactional',
+    messageId,
+    action: 'parseTransactionalEvent',
+  })
+
+  if (twilioErrorCode) {
+    await SmsMessageTransactional.update(
+      {
+        errorCode: twilioErrorCode,
+        status: TransactionalSmsMessageStatus.Error,
+      } as SmsMessageTransactional,
+      {
+        where: {
+          messageId: messageId,
+        },
+      }
+    )
+  } else {
+    // longer messages are delivered in multiple segments
+    // each segment has a separate delivery status
+    // Update the message as successful only if there does not exist previous failed status
+    // update only if sent or delivered
+    const mappedStatus = twilioMessageStatus.toUpperCase()
+    if (
+      mappedStatus == TransactionalSmsMessageStatus.Sent ||
+      mappedStatus == TransactionalSmsMessageStatus.Delivered
+    ) {
+      await SmsMessageTransactional.update(
+        {
+          updatedAt: new Date(),
+          status: twilioMessageStatus.toUpperCase(),
```

Comment:
I feel like we should have a function to cover twilioMessageStatus to our native txnmessagestatus then use across this handler

Reply:
used a switch case to map back, given that it's not a 1 to 1 relation i will only capture statuses which are the same, and the rest defaulted to unsent state